static/js/pages/App.js

```diff
@@ -19,50 +20,93 @@ import AccountSettingsPage from "./settings/AccountSettingsPage"
 import users, { currentUserSelector } from "../lib/queries/users"
 import { routes } from "../lib/urls"
 
+import { ALERT_TYPE_TEXT } from "../constants"
 import type { Match } from "react-router"
 
+import { addUserNotification } from "../actions"
+
+import type { CurrentUser } from "../flow/authTypes"
+
 type Props = {
-  match: Match
+  currentUser: CurrentUser,
+  match: Match,
+  addUserNotification: Function
 }
 
-export default function App(props: Props) {
-  const { match } = props
+export class App extends Component<Props> {
+  componentDidMount() {
+    const { addUserNotification } = this.props
+    const notifications = document.querySelector(".site-wide")
+    if (notifications) {
+      const notificationId = notifications.getAttribute("data-notification-id")
+      const notificationMessage = notifications.getElementsByTagName("p")[0]
+        .textContent
+      if (localStorage.getItem("dismissedNotification") !== notificationId) {
+        addUserNotification({
+          "cms-site-wide-notification": {
+            type:  ALERT_TYPE_TEXT,
+            props: {
+              text:        notificationMessage,
+              persistedId: notificationId
+            }
+          }
+        })
+      }
+    }
+  }
 
-  const currentUser = useSelector(currentUserSelector)
-  useRequest(users.currentUserQuery())
+  render() {
+    const { match, currentUser } = this.props
 
-  if (!currentUser) {
-    // application is still loading
-    return <div className="app" />
+    return (
+      <Fragment>
+        {currentUser ? (
```

Comment:
Earlier it was function-based components, now it has been converted to the class-based component so that type of rendering is not supported in class-based components.